src/uct/api/uct.h

```diff
@@ -427,6 +427,10 @@ typedef enum uct_atomic_op {
 #define UCT_IFACE_FLAG_TAG_EAGER_BCOPY UCS_BIT(51) /**< Hardware tag matching bcopy eager support */
 #define UCT_IFACE_FLAG_TAG_EAGER_ZCOPY UCS_BIT(52) /**< Hardware tag matching zcopy eager support */
 #define UCT_IFACE_FLAG_TAG_RNDV_ZCOPY  UCS_BIT(53) /**< Hardware tag matching rendezvous zcopy support */
+
+        /* Ordering flags */
+#define UCT_IFACE_FLAG_RELAXED_ORDERING UCS_BIT(54) /**< Interface can't guarantee data
```

Comment:
Do you think single flag is enough for all types of communication semantics ? For example, one transport can provide order for send/receive but no order for put/gets, etc.

Reply:
Also, what transport is going to be imitate user of this

Reply:
> Also, what transport is going to be imitate user of this

IB over adaptive routing

Reply:
> Do you think single flag is enough for all types of communication semantics ? For example, one transport can provide order for send/receive but no order for put/gets, etc.

It should be defined only for active messages. Put/get/atomic are already defined as weak order in UCT API.
Also, we may consider adding iface_open feature flag to "force" strict ordering (or to allow weak ordering if it improves BW)

Reply:
Can we go the other way around and assume that AM is relaxed by default and we will be adding strict order flag ? 
BTW, If this is just for AM, we have to make it AM specific.